src/hocon_schema.erl

```diff
@@ -275,11 +275,14 @@ fields(Sc, Name) ->
 -spec fields_and_meta(schema(), name()) -> fields().
 fields_and_meta(Mod, Name) when is_atom(Mod) ->
     Meta =
-        case Mod:fields(Name) of
+        try Mod:fields(Name) of
             Fields when is_list(Fields) ->
                 maybe_add_desc(Mod, Name, #{fields => Fields});
             Fields ->
                 ensure_struct_meta(Fields)
+        catch
+            error:function_clause ->
+                throw({missing_struct_def, Mod, Name})
```

Comment:
erlang:raise with original stackstace. 
function_clause may happen in deeper stack under Mod:fields